tests/api_fastapi/core_api/routes/public/test_assets.py

```diff
@@ -0,0 +1,190 @@
+        assert response.json()["detail"] == msg
+
+    @pytest.mark.parametrize(
+        "url, expected_assets",
+        [
+            ("/public/assets?uri_pattern=s3", {"s3://folder/key"}),
+            ("/public/assets?uri_pattern=bucket", {"gcp://bucket/key", "wasb://some_asset_bucket_/key"}),
+            (
+                "/public/assets?uri_pattern=asset",
+                {"somescheme://asset/key", "wasb://some_asset_bucket_/key"},
+            ),
+            (
+                "/public/assets?uri_pattern=",
+                {
+                    "gcp://bucket/key",
+                    "s3://folder/key",
+                    "somescheme://asset/key",
+                    "wasb://some_asset_bucket_/key",
+                },
+            ),
+        ],
+    )
+    @provide_session
+    def test_filter_assets_by_uri_pattern_works(self, test_client, url, expected_assets, session):
+        asset1 = AssetModel("s3://folder/key")
+        asset2 = AssetModel("gcp://bucket/key")
+        asset3 = AssetModel("somescheme://asset/key")
+        asset4 = AssetModel("wasb://some_asset_bucket_/key")
+
+        assets = [asset1, asset2, asset3, asset4]
+        for a in assets:
+            self.create_provided_asset(a)
+
+        response = test_client.get(url)
+        assert response.status_code == 200
+        asset_urls = {asset["uri"] for asset in response.json()["assets"]}
+        assert expected_assets == asset_urls
+
+
+class TestGetAssetsEndpointPagination(TestAssets):
+    @pytest.mark.parametrize(
+        "url, expected_asset_uris",
+        [
+            # Limit test data
+            ("/public/assets?limit=1", ["s3://bucket/key/1"]),
+            ("/public/assets?limit=100", [f"s3://bucket/key/{i}" for i in range(1, 101)]),
+            # Offset test data
+            ("/public/assets?offset=1", [f"s3://bucket/key/{i}" for i in range(2, 102)]),
+            ("/public/assets?offset=3", [f"s3://bucket/key/{i}" for i in range(4, 104)]),
+            # Limit and offset test data
+            ("/public/assets?offset=3&limit=3", [f"s3://bucket/key/{i}" for i in [4, 5, 6]]),
+        ],
+    )
+    def test_limit_and_offset(self, test_client, url, expected_asset_uris):
+        self.create_assets(110)
+
+        response = test_client.get(url)
+
+        assert response.status_code == 200
+        asset_uris = [asset["uri"] for asset in response.json()["assets"]]
+        assert asset_uris == expected_asset_uris
+
+    def test_should_respect_page_size_limit_default(self, test_client):
+        self.create_assets(110)
+
+        response = test_client.get("/public/assets")
+
+        assert response.status_code == 200
+        assert len(response.json()["assets"]) == 100
+
+    @conf_vars({("api", "maximum_page_limit"): "150"})
+    def test_should_return_conf_max_if_req_max_above_conf(self, test_client):
+        self.create_assets(200)
+
+        # change to 180 once format_parameters is integrated
```

Comment:
We do not have this so far, so I have hard coded the test to check for 150. Alternatively, we can get rid of this test and add it later too.

cc @pierrejeambrun what do you think?

Reply:
This is really straight forward to implement, you can just add a validation on the `LimitFilter` with `Query(le=....)`

Reply:
This will benefit to all endpoints using this query param

Reply:
I will take this in a follow up PR

Reply:
No problem, then I think we can remove the test because we cannot test what we want here. And you add it back in the following PR.